File: Backend/Models/PoliceEvent.cs
namespace Backend.Models;

public class PoliceEvent
{
    public int Id { get; set; }
    public required string Datetime { get; set; }
    public required string Name { get; set; }
    public required string Summary { get; set; }
    public required string Url { get; set; }
    public required string Type { get; set;  }
    public required Location Location { get; set; }
}

public class Location
{
    public required string Name { get; set; }
    public required string Gps { get; set; }
}

File: Frontend/src/Models/policeEvent.ts
export type EventResponse = {
  data: EventEntity[];
  message: string;
  status: boolean;
};

export type EventEntity = {
  id: number;
  eventDate: string;
  policeEvent: PoliceEvent;
};

export type PoliceEvent = {
  id: number;
  name: string;
  summary: string;
  url: string;
  type: string;
  location: Location;
};

export type Location = {
  name: string;
  gps: string;
};

export type EventIdResponse = {
  data: EventEntity;
  message: string;
  status: boolean;
};


